manifests/server/config.pp

```diff
@@ -201,15 +201,26 @@
       # - $service_name
       # - $port
       # - $datadir
+      file { 'systemd-conf-dir':
+        ensure => directory,
+        path   => "/etc/systemd/system/${service_name}.service.d",
+        owner   => root,
+        group   => root,
+        notify  => [ Exec['restart-systemd'], Class['postgresql::server::service'] ],
+        before  => Class['postgresql::server::reload'],
+      }
+
       file { 'systemd-override':
         ensure  => file,
-        path    => "/etc/systemd/system/${service_name}.service",
+        path    => "/etc/systemd/system/${service_name}.service.d/${service_name}.conf",
         owner   => root,
         group   => root,
         content => template('postgresql/systemd-override.erb'),
         notify  => [Exec['restart-systemd'], Class['postgresql::server::service']],
         before  => Class['postgresql::server::reload'],
+        require => File['systemd-conf-dir'],
```

Comment:
Isn't this autorequired since Puppet 2.7?

Reply:
I'm not certain about that.  I've had issues in the past with puppet trying to create files in a directory that doesn't exist which is why I added the requirement here.  If you want to leave it off that's fine though.

Reply:
In spec tests you can verify it:
```ruby
it { is_expected.to contain_file('/etc/systemd/postgresql-server.d/override.conf').that_requires('/etc/systemd/postgresql-server.d') }
```

Reply:
I just did a quick test and it appears that there is no autorequire for the parent directory.  The unit test fails after adding a test similar to above.